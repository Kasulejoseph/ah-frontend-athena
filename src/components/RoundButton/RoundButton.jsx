import React from "react";
import PropTypes from "prop-types";
import "./RoundButton.scss";
import { MDBIcon } from "mdbreact";

export default function RoundButton(props) {
  const { background } = props;
  const { onClick } = props;
  const { label } = props;
  const { hoverClass } = props;
  const { icon } = props;
  const classes = `my-round bg-${background} border relative d-flex flex-row m-1 p-2 align-items-center ${hoverClass}`;
  return (
    <div onClick={onClick} className={classes} role="presentation">
      <MDBIcon icon={icon} className="mr-2" />
      <p className="m-0">{label}</p>
    </div>
  );
}

RoundButton.propTypes = {
  onClick: PropTypes.func,
  background: PropTypes.string,
  label: PropTypes.string,
  icon: PropTypes.string,
  hoverClass: PropTypes.string
};

RoundButton.defaultProps = {
  onClick: () => {},
  background: "",
  label: "",
  icon: "",
  hoverClass: ""
};
