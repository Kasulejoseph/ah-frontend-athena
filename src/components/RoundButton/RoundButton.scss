.my-round {
  border-radius: 20px;
}

.hover {
  &:hover {
    border: 1px solid red !important;
    color: red;
  }
}
