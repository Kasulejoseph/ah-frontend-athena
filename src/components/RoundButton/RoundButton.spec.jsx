import React from "react";
import { shallow } from "enzyme";
import RoundButton from "./RoundButton";

let props = {
  onClick: jest.fn()
};

describe("test round button", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<RoundButton {...props} />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
