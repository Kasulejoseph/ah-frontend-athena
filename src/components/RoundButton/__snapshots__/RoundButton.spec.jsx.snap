// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test round button matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RoundButton
    background=""
    hoverClass=""
    icon=""
    label=""
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <t
          border={false}
          brand={false}
          className="mr-2"
          fab={false}
          fal={false}
          far={false}
          fixed={false}
          flip=""
          icon=""
          inverse=""
          light={false}
          list={false}
          pull=""
          pulse={false}
          regular={false}
          rotate=""
          size=""
          spin={false}
          stack=""
        />,
        <p
          className="m-0"
        >
          
        </p>,
      ],
      "className": "my-round bg- border relative d-flex flex-row m-1 p-2 align-items-center ",
      "onClick": [MockFunction],
      "role": "presentation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "border": false,
          "brand": false,
          "className": "mr-2",
          "fab": false,
          "fal": false,
          "far": false,
          "fixed": false,
          "flip": "",
          "icon": "",
          "inverse": "",
          "light": false,
          "list": false,
          "pull": "",
          "pulse": false,
          "regular": false,
          "rotate": "",
          "size": "",
          "spin": false,
          "stack": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "m-0",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <t
            border={false}
            brand={false}
            className="mr-2"
            fab={false}
            fal={false}
            far={false}
            fixed={false}
            flip=""
            icon=""
            inverse=""
            light={false}
            list={false}
            pull=""
            pulse={false}
            regular={false}
            rotate=""
            size=""
            spin={false}
            stack=""
          />,
          <p
            className="m-0"
          >
            
          </p>,
        ],
        "className": "my-round bg- border relative d-flex flex-row m-1 p-2 align-items-center ",
        "onClick": [MockFunction],
        "role": "presentation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "border": false,
            "brand": false,
            "className": "mr-2",
            "fab": false,
            "fal": false,
            "far": false,
            "fixed": false,
            "flip": "",
            "icon": "",
            "inverse": "",
            "light": false,
            "list": false,
            "pull": "",
            "pulse": false,
            "regular": false,
            "rotate": "",
            "size": "",
            "spin": false,
            "stack": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "m-0",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
